packages/boot/src/booters/datasource.booter.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {CoreBindings} from '@loopback/core';
+import {AppWithRepository, juggler, Class} from '@loopback/repository';
+import {inject} from '@loopback/context';
+import {kebabCase} from 'lodash';
+import {ArtifactOptions} from '../interfaces';
+import {BaseArtifactBooter} from './base-artifact.booter';
+import {BootBindings} from '../keys';
+
+/**
+ * A class that extends BaseArtifactBooter to boot the 'DataSource' artifact type.
+ * Discovered DataSources are bound using `app.controller()`.
+ *
+ * Supported phases: configure, discover, load
+ *
+ * @param app Application instance
+ * @param projectRoot Root of User Project relative to which all paths are resolved
+ * @param [bootConfig] DataSource Artifact Options Object
+ */
+export class DataSourceBooter extends BaseArtifactBooter {
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) public app: AppWithRepository,
+    @inject(BootBindings.PROJECT_ROOT) public projectRoot: string,
+    @inject(`${BootBindings.BOOT_OPTIONS}#datasources`)
+    public datasourceConfig: ArtifactOptions = {},
+  ) {
+    super();
+    // Set DataSource Booter Options if passed in via bootConfig
+    this.options = Object.assign({}, DataSourceDefaults, datasourceConfig);
+  }
+
+  /**
+   * Uses super method to get a list of Artifact classes. Boot each file by
+   * creating a DataSourceConstructor and binding it to the application class.
+   */
+  async load() {
+    await super.load();
+
+    /**
+     * If Repository Classes were discovered, we need to make sure RepositoryMixin
+     * was used (so we have `app.repository()`) to perform the binding of a
+     * Repository Class.
+     */
+    if (this.classes.length > 0) {
+      if (!this.app.dataSource) {
+        console.warn(
+          'app.dataSource() function is needed for DataSourceBooter. You can add ' +
+            'it to your Application using RepositoryMixin from @loopback/repository.',
+        );
+      } else {
+        this.classes.forEach(cls => {
+          const name = kebabCase(cls.name.replace('DataSource', ''));
```

Comment:
@bajtos This is where the name for binding the `DataSource` class gets changed from `DbDataSource` to `db`. 

Now that I look at it, maybe this would be better in the `.dataSource` function so anyone using it directly also benefits from this and not just boot users. Thoughts?

Reply:
IMO, this adds too much coupling between our core/repository APIs, the way how boot operates and the naming conventions we use in files generated by our tooling.

Ideally, the core & repository APIs should be agnostic of any class-naming conventions.

Also, what if we have `DbDataSource` class with JSON config file that says `name: "mysql-db"` (imagine we started with the same name but then a change was made in one of these two places only)?  Which name will take precedence? How are our users going to learn about these precedence rules? Isn't this making the learning curve more steeper than needed?